### PR TITLE
Goodie - Advanced Handle functions; adding documentation around css

### PR DESCRIPTION
**Pull request by mintsoft**
I noticed there's no documentation around autoslurping of stylesheets currently; I'm not sure if this is the best location for it, however it seemed to fit in.

@jagtalon @moollaza

### CODE REVIEW

File: duckduckhack/goodie/goodie_advanced_handle_functions.md [hunk 1]
```diff
@@ -34,16 +34,12 @@ Realistic examples are scattered through the Goodie Repository:
...
  * [URLDecode](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/Goodie/URLDecode.pm#L45..L50)
- 
+
 ## Using Files in the Share Directory
 
 Goodies can use simple text or html input files for display or processing. These files can be read once and reused to answer many queries without cluttering up your source code.
 
-The `share` function gives each Instant Answer access to a subdirectory of the repository's `share` directory. The subdirectory for your Instant Answer is based on its Perl package name which is transformed from CamelCase to underscore_separated_words.  For example, the [DDG::Goodie::RegexCheatSheet package](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/Goodie/RegexCheatSheet.pm) uses the directory [share/goodie/regex_cheat_sheet](https://github.com/duckduckgo/zeroclickinfo-goodies/tree/master/share/goodie/regex_cheat_sheet) to store its custom CSS.
-
-<!-- /summary -->
```
Comment (moollaza) [inline duckduckhack/goodie/goodie_advanced_handle_functions.md]: We still need this actually for the DuckDuckHack Longtail that parses the docs

File: duckduckhack/goodie/goodie_advanced_handle_functions.md [hunk 2]
```diff
@@ -77,3 +73,7 @@ As with the Passphrase Goodie example above, each line becomes an entry in the r
 In some cases, you may need to generate the data files you will `slurp` from the share directory. If so, please put the required generation scripts in the Goodie's `share` directory. While **shell scripts are preferred**, your scripts can be written in the language of your choice.
 
 As an example, the [CurrencyIn Goodie](https://github.com/duckduckgo/zeroclickinfo-goodies/tree/master/share/goodie/currency_in) uses a [combination of Shell and Python scripts](https://github.com/duckduckgo/zeroclickinfo-goodies/tree/master/share/goodie/currency_in) to generate its input data, `currency.txt`.
+
+## Stylesheets
+
+Goodies not using structured answers can use custom stylesheets from the `share` directory. A `.css` with the same filename as the directory is automatically included along with the Goodie's output. For example, the [DDG::Goodie::RegexCheatSheet package](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/Goodie/RegexCheatSheet.pm) uses the directory [share/goodie/regex_cheat_sheet] and has an automatically included stylesheet (https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/share/goodie/regex_cheat_sheet/regex_cheat_sheet.css).
```
Comment (moollaza) [inline duckduckhack/goodie/goodie_advanced_handle_functions.md]: Would slightly rephrase (and fix the broken markdown link):

Goodies can load custom stylesheets by adding a `.css` file to their `share` directory, with a filename identical to the parent directory. Goodie stylesheets will be automatically included along with the Goodie's output when triggered. For example, the [RegexCheatSheet Goodie](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/Goodie/RegexCheatSheet.pm) uses the directory 'share/goodie/regex_cheat_sheet' which contains a [custom stylesheet](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/share/goodie/regex_cheat_sheet/regex_cheat_sheet.css).

**Note:** Custom stylesheets should be used sparingly and only when absolutely necessary. 

Thoughts?

**Comment by moollaza** [comment]
@mintsoft thanks for doing this, made a couple suggestions

File: duckduckhack/goodie/goodie_advanced_handle_functions.md (hunk 1 above)
Comment (mintsoft) [inline duckduckhack/goodie/goodie_advanced_handle_functions.md]: The `<!-- /summary --> ` line or the entire block?

**Comment by mintsoft** [comment]
@moollaza No problem; I prefer your rewording. I've had to move a paragraph around to slot the `<!-- /summary -->` bit in

**Comment by moollaza** [comment]
LGTM :+1:

**Pull request merged**